exp/runtime/hooks/api/v1alpha1/topologymutation_variable_types.go

```diff
@@ -28,30 +28,39 @@ const BuiltinsName = "builtin"
 // Builtins represents builtin variables exposed through patches.
 type Builtins struct {
 	// cluster represents builtin cluster variables.
+	// +optional
 	Cluster *ClusterBuiltins `json:"cluster,omitempty"`
 	// controlPlane represents builtin ControlPlane variables.
+	// +optional
 	ControlPlane *ControlPlaneBuiltins `json:"controlPlane,omitempty"`
 	// machineDeployment represents builtin MachineDeployment variables.
+	// +optional
 	MachineDeployment *MachineDeploymentBuiltins `json:"machineDeployment,omitempty"`
 	// machinePool represents builtin MachinePool variables.
+	// +optional
 	MachinePool *MachinePoolBuiltins `json:"machinePool,omitempty"`
 }
 
 // ClusterBuiltins represents builtin cluster variables.
 type ClusterBuiltins struct {
 	// name is the name of the cluster.
```

Comment:
Let's keep this omitempty and the other further below please and make the fields optional accordingly (please check the entire file)

(It's a breaking change to make these fields required now)